### ISSUE TITLE
Missing argument for importing eigen in cmake

### ISSUE THREAD

**Issue by anuppari** [strasdat/Sophus#266]
The eigen library [suggests](https://eigen.tuxfamily.org/dox/TopicCMakeGuide.html) importing eigen with the `NO_MODULE` argument in `find_package`, i.e., [this line](https://github.com/strasdat/Sophus/blob/a0fe89a323e20c42d3cecb590937eb7a06b8343a/CMakeLists.txt#L35) should be
```cmake
find_package(Eigen3 3.3.0 REQUIRED NO_MODULE)
```
This enables consumers to choose the version of eigen via the `Eigen3_DIR` variable.

This fix may also address some of the issues in #187

**Comment by NikolausDemmel** [comment]
I don't think this is a good idea. Not providing `NO_MODULE` allows most flexibility by package users. They can then decide to provide a find module or use config mode.

Cmake's [own recommendation](https://cmake.org/cmake/help/latest/command/find_package.html#id4) is to prefer basic signature  (i.e. not pass `NO_MODULE`):

> Where possible, user code should generally look for packages using the [basic signature](https://cmake.org/cmake/help/latest/command/find_package.html#basic-signature), since that allows the package to be found with either mode.

Note that users of Sophus can already use the `Eigen3_DIR` or `Eigen3_ROOT` cmake variables to customize the Eigen location, even w/o `NO_MODULE`, e.g. by ensuring that cmake doesn't find a custom find module for Eigen so it uses config mode, or by passing [`CMAKE_FIND_PACKAGE_PREFER_CONFIG`](https://cmake.org/cmake/help/latest/variable/CMAKE_FIND_PACKAGE_PREFER_CONFIG.html#variable:CMAKE_FIND_PACKAGE_PREFER_CONFIG) to cmake to make it try config mode first.
